pkg/kubelet/eviction/testing/mock_memory_threshold_notifier.go

```diff
@@ -0,0 +1,52 @@
+/*
```

Comment:
Would it be worth writing some kind of stress test that checks for file descriptor leaks? It looks like there may be a lot of opportunities for leaks given how many different fds need to be managed by this feature.

Reply:
I am hoping this is caught by our soak tests...